docs/src/site/twiki/BacklogMetricEmitterService.twiki

```diff
@@ -0,0 +1,54 @@
+---+Back Log Emitter Service
+
+Back Log Emitter service keeps track of all the process which have missed an SLA for a particular time and calculates backlog. Once a notification comes that process has completed, it updates the backlog. This service keeps on sending data to Graphite for each process which is missing SLA.
+
+The emission and calculation of backlog are controlled by the below properties in startup.properties :
+
+<verbatim>
+*.falcon.backlog.metricservice.emit.interval.millisecs=60000
+*.falcon.backlog.metricservice.recheck.interval.millisecs=60000
+</verbatim>
+
+---+How Backlog is Calculated
+
+Back Log Emitter Service is a listener to :
+
+   1. EntitySLAAlertService
+   1. !WorkflowExecutionListener
+
+
+EntitySLAListeners service notifies !BacklogEmitter whenever a process misses its shouldEndIn sla.
+
+!WorkflowExecutionListener notifies !BacklogEmitter service whenever a process get completed.
+
+
+*Back Log Calculation:*
+
+If two instances of hourly process p1 have missed SLA for 4th and 5th hour then in 6th hourbacklog for process p1 will be ((6-4) + (6-5))*60 = 180 mins. It is cumulative in nature.
```

Comment:
Process p
then in the 6th hourbacklog -> then at 06:00 backlog  for the process would be shown as 180 mins(((6-4) + (6-5))*60)). It is cumulative by nature.